IbisLib/abstractpolydataobject.h

```diff
@@ -0,0 +1,158 @@
+/*=========================================================================
+Ibis Neuronav
+Copyright (c) Simon Drouin, Anna Kochanowska, Louis Collins.
+All rights reserved.
+See Copyright.txt or http://ibisneuronav.org/Copyright.html for details.
+
+     This software is distributed WITHOUT ANY WARRANTY; without even
+     the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+     PURPOSE.  See the above copyright notice for more information.
+=========================================================================*/
+#ifndef __AbstractPolyDataObject_h_
+#define __AbstractPolyDataObject_h_
+
+#include "sceneobject.h"
+#include "serializer.h"
+#include <map>
+#include <QVector>
+#include <QObject>
+#include <vtkProperty.h>
+#include <vtkSmartPointer.h>
+
+class vtkPolyData;
+class vtkTransform;
+class vtkActor;
+class vtkDataSetAlgorithm;
+class vtkImageData;
+class vtkProbeFilter;
+class vtkScalarsToColors;
+class ImageObject;
+class vtkClipPolyData;
+class vtkPassThrough;
+class vtkCutter;
+class vtkPlane;
+class vtkPlanes;
+
+enum RenderingMode{ Solid = 0, Wireframe = 1, Both = 2 };
+
+
+class AbstractPolyDataObject : public SceneObject
+{
+    
+Q_OBJECT
+
+public:
+        
+    //static AbstractPolyDataObject * New() { return new AbstractPolyDataObject; }
+    vtkTypeMacro(AbstractPolyDataObject,SceneObject);
+    
+    AbstractPolyDataObject();
+    virtual ~AbstractPolyDataObject();
+    virtual void Serialize( Serializer * ser ) override;
+    virtual void Export() override;
```

Comment:
I would implement Export() and IsExportable() in derived classes, we may have classes that will not be exportable, or that could be exported in different format.

Reply:
I moved it and tested the build